frontend/javascript/components/opensearch/opensearch-wrapper.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <SearchBar @search="handleSearch" class="mb-4" />
+  <div v-if="searchParams">
+    <template v-if="searchResults === null">
+       <div class="spinner-border" role="status"></div>
+    </template>
+    <template v-else-if="searchResults.length > 0">
+      <SearchResult v-for="result in searchResults" :key="result.id" v-bind="result" />
+    </template>
+    <div v-else>No results</div>
+    <div>
+      <button :disabled="searchParams.currentPage === 0" @click="searchParams.currentPage--">&lt;</button>
+      {{ searchParams?.currentPage + 1 }}
+      <button :disabled="searchParams.currentPage >= searchParams.maxPage"
+        @click="searchParams.currentPage++">&gt;</button>
+    </div>
+  </div>
+  <div v-else>Enter search term and click search</div>
+</template>
+
+<script lang="ts" setup>
+import SearchBar from './search-bar.vue'
+import SearchResult from './search-result.vue'
+import DOMPurify from 'dompurify';
+import { defineProps, ref, watch, onWatcherCleanup, Ref } from 'vue'
```

Comment:
[Note](https://vuejs.org/api/sfc-script-setup.html#defineprops-defineemits): `defineProps` and `defineEmits` are **compiler macros** only usable inside `<script setup>`. They do not need to be imported, and are compiled away when `<script setup>` is processed.